### DISCUSSION TITLE
How to create new file?

### DISCUSSION THREAD

**Discussion by jegatheesprana** [uptick/react-keyed-file-browser#195] [Q&A] [answered]
We can create new folder or sub folder. But how to create a file?

**Comment by dreadera** [comment, marked as answer]
> We can create new folder or sub folder. But how to create a file?

At this current stage, you cannot create a file in the same way you create a folder. However, you can create a component outside of the `FileBrowser` that mutates the file array thats passed to the `FileBrowser` which will allow this functionality :)

The idea of creating files within the library itself has been requested before, hence we do have it on our list of features we want to implement.

**Comment by jegatheesprana** [comment]
I looked at the source code and tried to add that feature. But I found that the necessary functions have already been developed. So I tried to add another action menu which will trigger required functions. But I ended up with some other issues.
If I click newly added button, it triggers the function but not showing the input box. Later I found that this is happen because globalClickEventListener triggers at that time.
Is this a problem you also face?

**Comment by jestrickler** [comment]
See https://github.com/uptick/react-keyed-file-browser/issues/59
